src/goose.rs

```diff
@@ -1109,26 +1124,26 @@ impl GooseClient {
                 }
             };
 
-            self.send_to_parent(raw_request);
+            self.send_to_parent(&raw_request);
         }
 
-        // @TODO: (improve comment) Consume request_name, if set.
+        // Consume request_name if set.
         if self.request_name != None {
             self.request_name = None;
         }
 
-        response
+        GooseResponse::new(raw_request, response)
```

Comment:
Probably need &response as you want a shared ref to it and not move it?

Reply:
we need to return it here, or we lose it